contracts/FundStorage.sol

```diff
@@ -68,7 +70,13 @@ contract FundStorage is Permissionable {
   struct MemberFines {
     uint256 total;
     // Assume ETH is address(0x1)
-    mapping(address => uint256) tokenFines;
+    mapping(address => MemberFineItem) tokenFines;
+  }
+
+  struct MemberFineItem {
+    uint256 amount;
+    uint256[] proposals;
+    address[] proposalsManagers;
```

Comment:
Here we have config contract. It's a really bad idea to cache values for UI here. This information could be kept in Proposal manager contract or any other contract.

Reply:
todo added